### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by kudukacar**
Add frontend to gilded rose application.  Please note frontend is not connected to backend yet.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
+frontend/.pnp.js
+
+# testing
+frontend/coverage
+
+# production
+frontend/build
+
+# misc
+frontend/.DS_Store
```
Comment (paytonrules) [inline .gitignore]: .DS_Store can be global. In fact you can even set that so it's never checked in on your machine.

https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

I use that for things like ~.DS_Store~ and emacs stuff.

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+        - cd backend
+      before_cache:
+        - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+        - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+      cache:
+        directories:
+          - $HOME/.gradle/caches/
+          - $HOME/.gradle/wrapper/
+    - name: "Frontend"
+      language: java
```
Comment (paytonrules) [inline .travis.yml]: I would look into Travis's built in JS builders, if that's available.

File: .gitignore (hunk 1 above)
Comment (kudukacar) [reply .gitignore]: Great thanks

File: .travis.yml (hunk 2 above)
Comment (kudukacar) [reply .travis.yml]: Yes I played around with the built in node.js language but kept getting errors.

Comment (kudukacar) [reply .travis.yml]: Hence the four failing commits above

Comment (kudukacar) [reply .travis.yml]: I finally figured out how to use the travis built in node.js language

**Review by damonkelley — APPROVED** [review]
🎉

**Review by hyoyou — APPROVED** [review]
🌟

**Pull request merged**